### DISCUSSION TITLE
Pagination of mongoDB with where clauses

### DISCUSSION THREAD

**Discussion by erkanisuf** [prisma/prisma#22846] [Q&A]
### Question

So i have this query:
`{
  skip: 0,
  take: 50,
  orderBy: { myField: 'asc' },
  where: { myField: 'myValue' }
} `
 

I have this pagination search with where clause. But when I put where `myFiels` is equal to `myValue` I am getting only results if the `skip` is set to `0`. But if `skip` is set to more than that no results. This is very annoying as if the Client sends `skip` and `take` values with filters it won't work, the only way to make it work is if there are `'where' `filters to make skip to: 0. I am not sure if there are other ways or am I missing something.

### How to reproduce (optional)

Use the query I provided with the basic schema model.

### Expected behavior (optional)

If there are filters/where clauses I should still get results no matter the skip value. 

### Information about Prisma Schema, Client Queries and Environment (optional)

```model BuildingRisk {
  id       String    @id @default(auto()) @map("_id") @db.ObjectId
  myField String
}
```

```ts
{
  skip: 0,
  take: 50,
  orderBy: { myField: 'asc' },
  where: { myField: 'myValue' }
} 
```

- OS: Mac/Windows/Any
- Database: MongoDB Atlas Cloud
- Node.js version: Any

- version 5.8 prisma

**Comment by nurul3101** [comment]
Hello @erkanisuf 👋 

If I understand correctly, you mean that you get the expected response when skip has the value of 0 (`skip:0`). 
But, if there is any other value, for example (`skip:10`), then you are not getting the expected response?

The skip parameter actuals skips the number of records that are specified. Can you please share a reproduction with a seed file, along with the expected response for each case? This would help in understanding your use case better.

**Reply by erkanisuf** [reply]
What I am trying to do is if the filter is there (especially equals to)  and there is already a record in DB regardless of the skip I want to get the record from the collection. 
I am attaching this test file which is JSON for mongoDB.
[TEST_FILE.json](https://github.com/prisma/prisma/files/14085344/TEST_FILE.json)

**Reply by ludralph** [reply]
Hi @erkanisuf 

Would this query work for you? For large datasets, it is recommended to use [cursor based pagination](https://www.prisma.io/docs/orm/prisma-client/queries/pagination#cursor-based-pagination)
```js
const firstPage = await prisma.yourModel.findMany({
  take: 50,
  where: { myField: 'myValue' },
  orderBy: { myField: 'asc' },
});

const lastItem = firstPage[firstPage.length - 1];
const nextPage = await prisma.yourModel.findMany({
  take: 50,
  where: { myField: 'myValue' },
  orderBy: { myField: 'asc' },
  cursor: { id: lastItem.id },
  skip: 1 // Skip the cursor
});
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
